### PR TITLE
add fileName to the output schema

### PR DESCRIPTION
**Pull request by gsps1**
adding filename field to the output schema.

### CODE REVIEW

**Comment by gsps1** [comment]
@bdmogal let me know if this should be in develop branch instead.

File: src/main/java/co/cask/WholeFileSource.java [hunk 1]
```diff
@@ -90,13 +90,16 @@ public String getInputFormatClassName() {
   }
 
   @Override
-  public void transform(KeyValue<NullWritable, BytesWritable> input,
+  public void transform(KeyValue<String, BytesWritable> input,
                         Emitter<StructuredRecord> emitter) throws Exception {
-    emitter.emit(StructuredRecord.builder(outputSchema).set("body", input.getValue().getBytes()).build());
+    emitter.emit(StructuredRecord.builder(outputSchema)
+                   .set("fileName", input.getKey())
```
Comment (chtyim) [inline src/main/java/co/cask/WholeFileSource.java]: Why just the name instead of the full path?

Comment (gsps1) [reply src/main/java/co/cask/WholeFileSource.java]: using full path now.

Comment (chtyim) [reply src/main/java/co/cask/WholeFileSource.java]: then you better call this "filePath"?

Comment (gsps1) [reply src/main/java/co/cask/WholeFileSource.java]: updated it

**Comment by chtyim** [comment]
LGTM

**Pull request merged**